### PR TITLE
Add `make clean` rule, don't have `include_dirs` nuke other things

### PR DESCRIPTION
**Pull request by Ericson2314**
I think this separation of concerns is good, though I don't need it for
Nixpkgs.

### CODE REVIEW

**Comment by Ericson2314** [comment]
@sbc100 this one good?

**Review by sbc100 — APPROVED** [review]
Nice!

File: Makefile [hunk 1]
```diff
@@ -420,7 +420,7 @@ $(LIBWASI_EMULATED_PROCESS_CLOCKS_OBJS): CFLAGS += \
     -I$(LIBC_BOTTOM_HALF_CLOUDLIBC_SRC)
 
 include_dirs:
-	$(RM) -r "$(SYSROOT)"
+	$(RM) -r "$(SYSROOT_INC)"
```
Comment (sbc100) [inline Makefile:423]: I think maybe we should completely remove this line?  If we are separating the cleaning from the building?

Comment (Ericson2314) [reply Makefile:423]: OK

Comment (Ericson2314) [reply Makefile]: I thought it would break `make; make`, but it doesn't!

**Review by sbc100 — APPROVED** [review]

**Comment by sbc100** [comment]
Side note: I wonder if renaming `$(SYSROOT)` to `$(STAGING_ROOT)` would make it more clear what is going on, and perhaps more acceptable to you in terms of terminology?

**Comment by Ericson2314** [comment]
@sbc100 I don't mind the names :). I may not like sysroots, but I do believe in alpha-equivalence :).

**Pull request merged**